sample-programs/erc20/erc20.scm

```diff
@@ -0,0 +1,129 @@
+      (fetch-entry allowances (tuple 
+        (owner owner) 
+        (spender spender))))))
+    (if (eq? allowance 'null) 0 allowance)))
+
+;; Credits balance of a specified principal.
+(define (credit-balance! (account principal) (amount int))
+  (if (<= amount 0)
+    'false
+    (let ((current-balance (balance-of account)))
+      (begin
+        (set-entry! balances 
+          (tuple (owner account))
+          (tuple (balance (+ amount current-balance)))) 
+        'true)))) ;; Overflow management?
+
+;; Debits balance of a specified principal.
+(define (debit-balance! (account principal) (amount int))
+  (let ((balance (balance-of account)))
+    (if (or (> amount balance) (<= amount 0))
+      'false
+      (begin
+        (set-entry! balances
+          (tuple (owner account))
+          (tuple (balance (- balance amount))))
+        'true))))
+
+;; Transfers tokens to a specified principal.
+(define (transfer! (sender principal) (recipient principal) (amount int))
+  (if (and (credit-balance! recipient amount) (debit-balance! sender amount)) 
+    'true
+    'false))
+
+;; Decrease allowance of a specified spender.
+(define (decrease-allowance! (spender principal) (owner principal) (amount int))
+  (let ((allowance (allowance-of spender owner)))
+    (if (or (> amount allowance) (<= amount 0))
+      'false
+      (begin
+        (set-entry! allowances
+          (tuple (spender spender) (owner owner))
+          (tuple (allowance (- allowance amount))))
+        'true))))
+
+;; Internal - Increase allowance of a specified spender.
+(define (increase-allowance! (spender principal) (owner principal) (amount int))
+  (let ((allowance (allowance-of spender owner)))
+    (if (<= amount 0)
+      'false
+      (begin
+        (set-entry! allowances 
+          (tuple (spender spender) (owner owner))
+          (tuple (allowance (+ allowance amount))))
+        'true)))) 
+
+;; Public functions
+
+;; Transfers tokens to a specified principal.
+(define-public (transfer (recipient principal) (amount int))
+  (transfer! tx-sender recipient amount))
+
+;; Transfers tokens to a specified principal, performed by a spender
+(define-public (transfer-from (owner principal) (recipient principal) (amount int))
+  (let ((allowance (allowance-of tx-sender owner)))
+    (if (or (> amount allowance) (<= amount 0))
+      'false
+      (begin
+        (transfer! owner recipient amount)
+        (decrease-allowance! tx-sender owner amount)))))
+
+;; Update the allowance for a given spender
+(define-public (approve (spender principal) (amount int))
+  (if (<= amount 0)
+    'false
+    (begin
+      (increase-allowance! spender tx-sender amount))))
+
+;; Mint new tokens.
+(define-public (mint! (account principal) (amount int))
+  (if (<= amount 0)
+    'false
+    (let ((balance (balance-of account)))
+      (begin 
+        (set-entry! balances
+          (tuple (owner account))
+          (tuple (balance (+ balance amount))))
+        'true))))

```

Comment:
Could you add some commented-out code that shows how to initialize the `balances` and `allowances` tables so that, at the very least, the contract creators get some tokens that they may transfer to other users?

Reply:
Related to https://github.com/blockstack/blockstack-core/pull/955/files/a15254e48505275ec7c571823c722ed23f6a1173#discussion_r285348131

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/955/commits/64b2652214b20518a539c910648e2a3d02f0ee93